### PR TITLE
Add ssh troubleshooting guide

### PR DESCRIPTION
**Pull request by mathsaey**
Based on the suggestions on the irc channel, I added a page to the docs which discusses some common issues and how to solve them. I wrote a section in this guide about solving common ssh related issues.

### CODE REVIEW

File: content/doc/help/troubleshooting.en-us.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  * Try to `ssh git@myremote.example` to ensure that everything is set up
+    properly.
+* On the server:
+  * Check the permissions of the `.ssh` directory in the home directory of your
+    `git` user.
+  * Verify that the correct public keys are added to `.ssh/authorized_keys`.
+    Try to run `Rewrite '.ssh/authorized_keys' file (for Gitea SSH keys)` on the
+    Gitea admin panel.
+
+If you get a similar error, but without the public key part (shown below),
```
Comment (lafriks) [inline content/doc/help/troubleshooting.en-us.md]: there should be no comma before `but`. Also two 'but' in same sentence is not good style and makes it hard to understand. Probably removing comma after `(shown below)` and adding `then` in it's place would be better

Comment (mathsaey) [reply content/doc/help/troubleshooting.en-us.md]: Good points; I made some minor changes to the text based on your feedback.

**Comment by lunny** [comment]
LGTM

**Review by appleboy — APPROVED** [review]

**Comment by appleboy** [comment]
LGTM

**Pull request merged**